hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4326,33 +4328,59 @@ func (r *HostedClusterReconciler) reconcileKubevirtPlatformDefaultSettings(ctx c
 	if hc.Spec.Platform.Kubevirt.GenerateID == "" {
 		hc.Spec.Platform.Kubevirt.GenerateID = utilrand.String(10)
 	}
-
 	// auto generate the basedomain by retrieving the default ingress *.apps dns.
 	if hc.Spec.Platform.Kubevirt.BaseDomainPassthrough != nil && *hc.Spec.Platform.Kubevirt.BaseDomainPassthrough {
-		// get base domain from default ingress of management cluster
-		defaultIngressOperator := manifests.IngressDefaultIngressController()
-		err := r.Client.Get(ctx, client.ObjectKeyFromObject(defaultIngressOperator), defaultIngressOperator)
-		if err != nil {
-			return err
+		// get base domain from default ingress of management/infra cluster
+		var targetClient client.Client
+		// If external infra is used, generate the infra client. Otherwise, use the client of the mgmt cluster
+		var namespace string
+		if hc.Spec.Platform.Kubevirt.Credentials != nil {
+			var err error
+			targetClient, err = generateInfraClusterClient(ctx, r.Client, *hc)
```

Comment:
maybe for clarity, this function should be renamed from generateInfraClusterClient to generateKubevirtInfraClusterClient()